source/proof_document/spec/proof_document_spec.rb

```diff
@@ -0,0 +1,235 @@
+
+
+    s3_client = Aws::S3::Client.new(stub_responses: true)
+    s3_client.stub_responses(:get_object, { body: "\xDE\x9E[\xF8\xB8\xABZ\xD2E\xA8\xC5`'\x18\xAF\xC7" })
+    allow(Aws::S3::Client).to receive(:new).and_return(s3_client)
+    allow_any_instance_of(IdentityDocAuth::Acuant::Responses::GetResultsResponse).to receive(:pii_from_doc).and_return(applicant_pii)
+  end
+
+  describe '.handle' do
+    before do
+      stub_request(
+        :post,
+        'https://lexisnexis.example.com/restws/identity/v2/abc123/aaa/conversation'
+      ).to_return(
+        body: {
+          "Status" => {
+            "TransactionStatus" => "passed"
+          }
+        }.to_json
+      )
+
+      stub_request(:post, callback_url).
+        with(
+          headers: {
+            'Content-Type' => 'application/json',
+            'X-API-AUTH-TOKEN' => idp_api_auth_token,
+          },
+        ) do |request|
+          expect(JSON.parse(request.body, symbolize_names: true)).to eq(
+            document_result: {
+              acuant_error:{ code:nil, message: nil}, billed: true, errors:{},
+              liveness_assessment: 'Live',
+              liveness_score: nil,
+              match_score: nil,
+              raw_alerts: [],
+              result: 'Passed',
+              success: true,
+              exception: nil,
+            }
+          )
+        end
+    end
+
+    it 'runs' do
+      IdentityIdpFunctions::ProofDocument.handle(event: event, context: nil)
+    end
+
+    context 'when called with a block' do
+      it 'gives the results to the block instead of posting to the callback URL' do
+        yielded_result = nil
+        IdentityIdpFunctions::ProofDocument.handle(
+          event: event,
+          context: nil
+        ) do |result|
+          yielded_result = result
+        end
+
+        expect(yielded_result).to eq(
+          document_result: {
+            acuant_error:{ code:nil, message: nil}, billed: true, errors:{},
+            liveness_score: nil,
+            match_score: nil,
+            raw_alerts: [],
+            result: 'Passed',
+            success: true,
+            liveness_assessment: 'Live',
+            exception: nil,
+          }
+        )
+
+        expect(a_request(:post, callback_url)).to_not have_been_made
+      end
+    end
+  end
+
+  describe '#proof' do
+    subject(:function) do
+      IdentityIdpFunctions::ProofDocument.new(event)
+    end
+
+    let(:document_proofer) { instance_double(IdentityDocAuth::Acuant::AcuantClient) }
+
+    before do
+      allow(function).to receive(:document_proofer).and_return(document_proofer)
+
+      stub_request(:post, callback_url).
+        with(headers: { 'X-API-AUTH-TOKEN' => idp_api_auth_token })
+    end
+
+    context 'with a successful response from the proofer' do
+      before do
+        expect(document_proofer).to receive(:post_images).
+          and_return(Proofer::Result.new)
```

Comment:
the `identity-doc-auth` gem doesn't use `Proofer::Result` at all, it uses its own response class `IdentityDocAuth::Response`, can we use that instead?